source/CILobbyMenu.cpp

```diff
@@ -275,19 +278,19 @@ void LobbyMenu::update(MenuState& state, string& joingame, string& playername, s
             if (_lobbyRoomLabel->getText().empty()) {
                 // TODO: update lobby room label using room id from network connection
             }
-            if (othernames.size() == 4) {
+            if (_otherNames.size() == 4) {
                 // update other players' labels
-                _gamelobbyplayerlabel2->setText(othernames.at(0));
-                _gamelobbyplayerlabel3->setText(othernames.at(1));
-                _gamelobbyplayerlabel4->setText(othernames.at(2));
-                _gamelobbyplayerlabel5->setText(othernames.at(3));
+                _gamelobbyplayerlabel2->setText(_otherNames.at(0));
+                _gamelobbyplayerlabel3->setText(_otherNames.at(1));
+                _gamelobbyplayerlabel4->setText(_otherNames.at(2));
+                _gamelobbyplayerlabel5->setText(_otherNames.at(3));
             }
 
             // Update game settings in global MenuScene  
-            spawnRate = _lobbySpawnRate;
-            gravStrength = _lobbyGravityStrength;
-            colorCount = _lobbyColorCount;
-            winPlanetMass = _lobbyWinPlanetMass;
+            gameSettings->setSpawnRate(_lobbySpawnRate);
```

Comment:
You should not be setting these variables in update. Each menu scene class should have a pointer to gameSettings and/or playerSettings so that when one of them gets updated all of the classes have an updated value